x/warp/config.go

```diff
@@ -0,0 +1,213 @@
+// that disables Warp.
+func NewDisableConfig(blockTimestamp *big.Int) *Config {
+	return &Config{
+		Upgrade: precompileconfig.Upgrade{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+
+// Key returns the key for the Warp precompileconfig.
+// This should be the same key as used in the precompile module.
+func (*Config) Key() string { return ConfigKey }
+
+// Verify tries to verify Config and returns an error accordingly.
+func (c *Config) Verify() error {
+	if c.QuorumNumerator > params.WarpQuorumDenominator {
+		return fmt.Errorf("cannot specify quorum numerator (%d) > quorum denominator (%d)", c.QuorumNumerator, params.WarpQuorumDenominator)
+	}
+	// If a non-default quorum numerator is specified and it is less than the minimum, return an error
+	if c.QuorumNumerator != 0 && c.QuorumNumerator < params.WarpQuorumNumeratorMinimum {
+		return fmt.Errorf("cannot specify quorum numerator (%d) < min quorum numerator (%d)", c.QuorumNumerator, params.WarpQuorumNumeratorMinimum)
+	}
+	return nil
+}
+
+// Equal returns true if [s] is a [*Config] and it has been configured identical to [c].
+func (c *Config) Equal(s precompileconfig.Config) bool {
+	// typecast before comparison
+	other, ok := (s).(*Config)
+	if !ok {
+		return false
+	}
+	equals := c.Upgrade.Equal(&other.Upgrade)
+	return equals && c.QuorumNumerator == other.QuorumNumerator
+}
+
+func (c *Config) Accept(acceptCtx *precompileconfig.AcceptContext, txHash common.Hash, logIndex int, topics []common.Hash, logData []byte) error {
+	unsignedMessage, err := warp.ParseUnsignedMessage(logData)
+	if err != nil {
+		return fmt.Errorf("failed to parse warp log data into unsigned message (TxHash: %s, LogIndex: %d): %w", txHash, logIndex, err)
+	}
+	log.Info("Accepted warp unsigned message", "txHash", txHash, "logIndex", logIndex, "logData", common.Bytes2Hex(logData))
+	if err := acceptCtx.Warp.AddMessage(unsignedMessage); err != nil {
+		return fmt.Errorf("failed to add warp message during accept (TxHash: %s, LogIndex: %d): %w", txHash, logIndex, err)
+	}
+	return nil
+}
+
+// verifyWarpMessage checks that [warpMsg] can be parsed as an addressed payload and verifies the Warp Message Signature
+// within [predicateContext].
+func (c *Config) verifyWarpMessage(predicateContext *precompileconfig.ProposerPredicateContext, warpMsg *warp.Message) error {
+	// Use default quourum numerator unless config specifies a non-default option
+	quorumNumerator := params.WarpDefaultQuorumNumerator
+	if c.QuorumNumerator != 0 {
+		quorumNumerator = c.QuorumNumerator
+	}
+
+	// Verify the warp payload can be decoded to the expected type
+	_, err := warpPayload.ParseAddressedPayload(warpMsg.UnsignedMessage.Payload)
+	if err != nil {
+		return fmt.Errorf("%w: %s", errInvalidAddressedPayload, err)
+	}
+
+	log.Info("verifyingWarpMessage", "warpMsg", warpMsg, "quorumNum", quorumNumerator, "quorumDenom", params.WarpQuorumDenominator)
```

Comment:
```suggestion
	log.Debug("verifying warp message", "warpMsg", warpMsg, "quorumNum", quorumNumerator, "quorumDenom", params.WarpQuorumDenominator)
```
I think should be debug since invalid txs could spam this log?